java-monitoring/run_protoc.sh

```diff
@@ -0,0 +1,128 @@
+#####################################################
+"${PROTOC_ROOT}"/protoc --experimental_allow_proto3_optional \
+"--plugin=protoc-gen-java_gapic=${REPO_ROOT}/gapic-generator-java-wrapper" \
+"--java_gapic_out=metadata:${LIBRARY_GEN_OUT}/google/monitoring/v3/monitoring_java_gapic_srcjar_raw.srcjar.zip" \
+"--java_gapic_opt=transport=grpc,rest-numeric-enums,grpc-service-config=google/monitoring/v3/monitoring_grpc_service_config.json,gapic-config=google/monitoring/v3/monitoring_gapic.yaml,api-service-config=google/monitoring/v3/monitoring.yaml" \
+${PROTO_FILES} google/cloud/common_resources.proto
+
+unzip -o -q "${LIBRARY_GEN_OUT}"/google/monitoring/v3/monitoring_java_gapic_srcjar_raw.srcjar.zip -d "${LIBRARY_GEN_OUT}"/google/monitoring/v3/
+# Sync'\''d to the output file name in Writer.java.
+unzip -o -q "${LIBRARY_GEN_OUT}"/google/monitoring/v3/temp-codegen.srcjar -d "${LIBRARY_GEN_OUT}"/google/monitoring/v3/monitoring_java_gapic_srcjar
+# Resource name source files.
+PROTO_DIR=${LIBRARY_GEN_OUT}/google/monitoring/v3/monitoring_java_gapic_srcjar/proto/src/main/java
+if [ ! -d "${PROTO_DIR}" ]
+then
+  # Some APIs don'\''t have resource name helpers, like BigQuery v2.
+  # Create an empty file so we can finish building. Gating the resource name rule definition
+  # on file existences go against Bazel'\''s design patterns, so we'\''ll simply delete all empty
+  # files during the final packaging process (see java_gapic_pkg.bzl)
+  mkdir -p "${PROTO_DIR}"
+  touch "${PROTO_DIR}"/PlaceholderFile.java
+fi
+cd "${LIBRARY_GEN_OUT}"/google/monitoring/v3/monitoring_java_gapic_srcjar/proto/src/main/java
+zip -r "${LIBRARY_GEN_OUT}"/monitoring_java_gapic_srcjar-resource-name.srcjar ./
+
+cd "${LIBRARY_GEN_OUT}"
+# Main source files.
+mkdir -p "${LIBRARY_GEN_OUT}"/google/monitoring/v3/google-cloud-monitoring-v3-java/gapic-google-cloud-monitoring-v3-java/src/main
+cp -r "${LIBRARY_GEN_OUT}"/google/monitoring/v3/monitoring_java_gapic_srcjar/src/main/java "${LIBRARY_GEN_OUT}"/google/monitoring/v3/google-cloud-monitoring-v3-java/gapic-google-cloud-monitoring-v3-java/src/main
+rm -r -f "${LIBRARY_GEN_OUT}"/google/monitoring/v3/google-cloud-monitoring-v3-java/gapic-google-cloud-monitoring-v3-java/src/main/java/META-INF
+# Remove empty files. If there are no resource names, one such file might have
+# been created. See java_gapic.bzl.
+find "${LIBRARY_GEN_OUT}"/google/monitoring/v3/google-cloud-monitoring-v3-java/gapic-google-cloud-monitoring-v3-java/src/main/java -type f -size 0 | while read f; do rm -f "${f}"; done
+
+if [ -d "${LIBRARY_GEN_OUT}"/google/monitoring/v3/google-cloud-monitoring-v3-java/gapic-google-cloud-monitoring-v3-java/src/main/java/samples ]; then
+    mv "${LIBRARY_GEN_OUT}"/google/monitoring/v3/google-cloud-monitoring-v3-java/gapic-google-cloud-monitoring-v3-java/src/main/java/samples "${LIBRARY_GEN_OUT}"/google/monitoring/v3/google-cloud-monitoring-v3-java/gapic-google-cloud-monitoring-v3-java
+fi
```

Comment:
This section is repeated below. Can we deduplicate using a function or something?